packages/dynamite/dynamite/lib/src/models/type_result/some_of.dart

```diff
@@ -0,0 +1,159 @@
+part of 'type_result.dart';
+
+@immutable
+abstract class TypeResultSomeOf extends TypeResult {
+  TypeResultSomeOf(
+    super.className, {
+    required this.subTypes,
+    super.nullable,
+  }) : assert(subTypes.isNotEmpty, 'Must have at least one subType');
+
+  @protected
+  final BuiltSet<TypeResult> subTypes;
+
+  @override
+  bool get isTypeDef => isSingleValue;
+
+  @override
+  String? get _builderFactory => null;
+
+  late final bool isSingleValue = optimizedSubTypes.length == 1;
+
+  @override
+  String? get _serializer {
+    if (isSingleValue) {
+      return null;
+    }
+
+    return '..add(${typeName}Extension.serializer)';
+  }
+
+  late final BuiltSet<TypeResult> optimizedSubTypes = _optimizedSubTypes.toBuiltSet();
+
+  @override
+  TypeResult get dartType {
+    if (isSingleValue) {
+      return optimizedSubTypes.single;
+    }
+
+    final record = optimizedSubTypes.map((final type) {
+      final dartType = type.nullableName;
+      final dartName = toDartName(dartType);
+      final fieldName = toFieldName(dartName, type.className);
+
+      return '$dartType $fieldName';
+    });
+
+    return TypeResultBase('({${record.join(',')}})');
+  }
+
+  late final String typeName = _typeName;
+
+  String get _typeName {
+    final buffer = StringBuffer(r'$');
+    for (final type in optimizedSubTypes) {
+      buffer.write(toDartName(type.className, uppercaseFirstCharacter: true));
+    }
+
+    return buffer.toString();
+  }
+
+  BuiltList<TypeResult> get _optimizedSubTypes {
+    final subTypes = BuiltSet<TypeResult>.build((final b) {
```

Comment:
```suggestion
    final flattenedSubTypes = BuiltSet<TypeResult>.build((final b) {
```

Better to indicate what this variable actually contains.

Reply:
Hm I don't know if that is a good idea or a bad one. We can leave it as is for now.